devicetypes/smartthings/ct100-thermostat.src/ct100-thermostat.groovy

```diff
@@ -624,7 +624,8 @@ def ping() {
 def switchMode() {
 	def currentMode = device.currentValue("thermostatMode")
 	def supportedModes = state.supportedModes
-	if (supportedModes) {
+	// Old version of supportedModes was as string, make sure it gets updated
+	if (supportedModes && supportedModes[0].size() > 1) {
```

Comment:
Is it possible for supportedModes to be an empty list, where supportedModes[0] would still throw an NPE?